### PR TITLE
spread: cryptography==3.4 and no rust for python_legacy

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Review by cjp256 — COMMENTED** [review]
```
+ pip3 install --user --upgrade cryptography==3.4 wheel setuptools pip==20.1
Collecting cryptography==3.4
  Downloading https://files.pythonhosted.org/packages/ea/d8/2afd2890fe451a3c109d2bdb6bc4ded55ec43059e524344d5e0004e36412/cryptography-3.4.tar.gz (544kB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-hkhz9emg/cryptography/setup.py", line 13, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'
```

crypto has to come over pip/wheel/setuptools

File: spread.yaml [hunk 1]
```diff
@@ -497,11 +497,11 @@ suites:
      - ubuntu-18.04*
    environment:
      SNAPCRAFT_PACKAGE_TYPE: "snap"
+     CRYPTOGRAPHY_DONT_BUILD_RUST: "1"
    prepare: |
      apt-get update
      apt-get install --yes gcc g++ make python3-dev python3-pip python3-wheel libffi-dev libsodium-dev libapt-pkg-dev squashfs-tools xdelta3 bzr git mercurial subversion libxml2-dev libxslt-dev
-     pip3 install --user --upgrade wheel setuptools
-     pip3 install --user --upgrade pip==20.1
+     pip3 install --user --upgrade cryptography==3.4 wheel setuptools pip==20.1
```
Comment (cjp256) [inline spread.yaml:504]:
```suggestion
     pip3 install --user --upgrade wheel setuptools
     pip3 install --user --upgrade pip==20.1
     pip3 install --user --upgrade cryptography==3.4
```

Comment (sergiusens) [reply spread.yaml]: But this will break, setuptools depends on cryptography...

**Comment by sergiusens** [comment]
not required anymore

**Pull request closed without merging**